SystemInformer/mainwnd.c

```diff
@@ -100,17 +100,84 @@ BOOLEAN PhMainWndInitialization(
+        if (!PhGetIntegerSetting(L"EnableVerboseWindowTitle"))
         {
-            PhAppendStringBuilder2(&stringBuilder, L" [");
-            PhAppendStringBuilder(&stringBuilder, &currentUserName->sr);
-            PhAppendCharStringBuilder(&stringBuilder, L']');
-            if (KphIsConnected()) PhAppendCharStringBuilder(&stringBuilder, L'+');
-            PhDereferenceObject(currentUserName);
+            if (currentUserName = PhGetSidFullName(PhGetOwnTokenAttributes().TokenSid, TRUE, NULL))
+            {
+                PhAppendStringBuilder2(&stringBuilder, L" [");
+                PhAppendStringBuilder(&stringBuilder, &currentUserName->sr);
+                PhAppendCharStringBuilder(&stringBuilder, L']');
+                if (KphIsConnected()) PhAppendCharStringBuilder(&stringBuilder, L'+');
+                PhDereferenceObject(currentUserName);
+            }
+
+            if (PhGetOwnTokenAttributes().ElevationType == TokenElevationTypeFull)
+                PhAppendStringBuilder2(&stringBuilder, L" (Administrator)");
         }
+        else
+        {
+            DWORD myPID = GetCurrentProcessId();
+            WCHAR myPIDStrArr[PH_INT32_STR_LEN_1];
+            PhPrintUInt32(myPIDStrArr, (ULONG)myPID);
+            PhAppendStringBuilder2(&stringBuilder, L" ");
+            PhAppendStringBuilder2(&stringBuilder, myPIDStrArr);
+            PhAppendStringBuilder2(&stringBuilder, L" ");
+
+            BOOL isRunningAsSystem = FALSE;
+
+            if (currentUserName = PhGetSidFullName(PhGetOwnTokenAttributes().TokenSid, TRUE, NULL))
+            {
+                PPH_STRING strSystemUsername = PhCreateString(L"NT Authority\\System");
+                isRunningAsSystem = (PhCompareString(currentUserName, strSystemUsername, TRUE) == 0);
+                PhDereferenceObject(strSystemUsername);
+
+                if (KphIsConnected())
+                    PhAppendStringBuilder2(&stringBuilder, L" <Driver-Connected>");
+                else
+                    PhAppendStringBuilder2(&stringBuilder, L" <No-Driver>");
+
+                PhDereferenceObject(currentUserName);
+            }
+
+            // Session ID\WindowStation\Desktop ...
 
-        if (PhGetOwnTokenAttributes().ElevationType == TokenElevationTypeFull)
-            PhAppendStringBuilder2(&stringBuilder, L" (Administrator)");
+            PhAppendStringBuilder2(&stringBuilder, L" : ");
+            ULONG sessionID;
+            PhGetProcessSessionId(GetCurrentProcess(), &sessionID);
+
+            WCHAR SessionIdString[PH_INT32_STR_LEN_1];
+            PhPrintUInt32(SessionIdString, sessionID);
+            PhAppendStringBuilder2(&stringBuilder, SessionIdString);
+            PhAppendStringBuilder2(&stringBuilder, L"\\");
+
+            // Read the desktop name of this current process -- This will suffice for now
+            //      Would instead be better to open the handle to this desktop and Read the object name, since this string could be blank
```

Comment:
Is there a reason you didn't do this? What you have here seems reasonable to me.

Reply:
Wasn’t sure if this was overkill, or not. I guess there is also some(?) value in seeing what was passed to a CreateProcess[xxx] function for the WinSta\Desktop.

But yes, most accurate to do ObjectNameInformation for the desktop handle.